### PR TITLE
[SW-1711] Fix propagation of internal security conf in Sparkling Water

### PR DESCRIPTION
**Pull request by jakubhava**
Depends on https://github.com/h2oai/h2o-3/pull/4073

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala [hunk 1]
```diff
@@ -50,14 +52,16 @@ class ExternalH2OBackend(val hc: H2OContext) extends SparklingBackend with Exter
...
     conf.sslConf match {
-      case Some(ssl) =>
-        val sslConfig = new Properties()
-        sslConfig.load(new FileInputStream(ssl))
-        cmdToLaunch = cmdToLaunch ++ Array("-files", sslConfig.get("h2o_ssl_jks_internal") + "," + sslConfig.get("h2o_ssl_jts"))
-        cmdToLaunch = cmdToLaunch ++ Array("-internal_security", ssl)
-        logInfo(s"Running external H2O cluster in encrypted mode with config: $ssl")
+      case Some(internalSecurityConf) =>
+      // In External Backend, auto mode we need distribute the keystore files to the H2O cluster
+      val props = new Properties()
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: nit: indentation

File: core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala [hunk 2]
```diff
@@ -60,6 +63,14 @@ private[backends] trait SharedBackendUtils extends Logging with Serializable {
       AzureDatabricksUtils.setClientCheckRetryTimeout(conf)
     }
 
+    if (conf.isInternalSecureConnectionsEnabled) {
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala]: Just a question: Is this executed only on a client? If so, can you point me to the place where it's set for workers?

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala (hunk 1 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala]: thanks!

File: core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala (hunk 2 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/SharedBackendUtils.scala]: Yup, it is executed only on client -> we create the configuration object on the client and then distribute the files to spark

**Pull request merged**